src/amo/components/VPNPromoBanner/index.js

```diff
@@ -98,6 +115,9 @@ export class VPNPromoBannerBase extends React.Component<InternalProps> {
       label: String(impressionCount),
     });
     _localStorage.removeItem(IMPRESSION_COUNT_KEY);
+    // See https://github.com/mozilla/addons-frontend/issues/10770
+    this._updateVariant(NOT_IN_EXPERIMENT);
+    this.setState({ showPromo: false });
```

Comment:
Why is a local state needed?

Reply:
Ok I see now, that's because there is no re-render when the user clicks the dismiss button. I think that's a valid use of a local state and we don't need `withUIState` here. Maybe we could only capture the dismiss click in the state, though.

I'd introduce a state like this:

```js
this.state = { wasDismissed: false };
```

The `this.setState()` call above should stay there and be updated accordingly.

Last, I'd introduce an early return in `shouldShowBanner`:

```js
  shouldShowBanner(): boolean {
    const { clientApp, regionCode, variant } = this.props;

    // User clicked the dismiss button.
	if (this.state.wasDismissed) {
      return false;
    }

    return (
      variant === VARIANT_SHOW &&
      !shouldExcludeUser({
        clientApp,
        regionCode,
      })
    );
  }
```

Reply:
Good suggestions. I have incorporated them into the next commit.

Reply:
Actually, while making these changes, it seems that `showPromo` is actually a better name for this state variable than `wasDismissed`, because it also deals with the value of `variant`. Take a look at the latest patch to see what I mean.

Reply:
I think it'd be best to not handle the `variant` in the state because that's a prop, which could potentially be updated and there is no code to sync the state. That being said, we both know that the prop value isn't going to change in _this_ case. Yet, I still think it'd be best to have a local state only for the dismissal but I won't fight for it ;-)

Reply:
Fair enough. It's true that it's not really a proper use of the state, although it's convenient. I'll change it back.